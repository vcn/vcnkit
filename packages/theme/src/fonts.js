export default {
    heroFamily: '"Roboto", sans-serif',
    heroSize: 2.125,
    heroWeight: 400,
    heroLineHeight: 1.4,

    headlineFamily: '"Roboto", sans-serif',
    headlineSize: 1.5,
    headlineWeight: 400,
    headlineLineHeight: 1.4,

    title1Family: '"Roboto", sans-serif',
    title1Size: 1.25,
    title1Weight: 400,
    title1LineHeight: 1.4,

    title2Family: '"Roboto", sans-serif',
    title2Size: 1.25,
    title2Weight: 500,
    title2LineHeight: 1.4,

    subheading1Family: '"Roboto", sans-serif',
    subheading1Size: 0.9375,
    subheading1Weight: 400,
    subheading1LineHeight: 1.4,

    subheading2Family: '"Roboto", sans-serif',
    subheading2Size: 0.9375,
    subheading2Weight: 500,
    subheading2LineHeight: 1.4,

    body1Family: '"Roboto", sans-serif',
    body1Size: 0.8125,
    body1Weight: 400,
    body1LineHeight: 1.618,

    body2Family: '"Roboto", sans-serif',
    body2Size: 0.8125,
    body2Weight: 500,
    body2LineHeight: 1.618,

    captionFamily: '"Roboto", sans-serif',
    captionSize: 0.75,
    captionWeight: 400,
    captionLineHeight: 1.618,
};
