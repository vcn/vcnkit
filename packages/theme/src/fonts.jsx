export default {
    heroFamily: '"Roboto", sans-serif',
    heroSize: 3,
    heroWeight: 400,
    heroLineHeight: 1.04166666,
    heroLetterSpacing: 0.003,

    title1Family: '"Roboto", sans-serif',
    title1Size: 2.125,
    title1Weight: 400,
    title1LineHeight: 1.20588235,
    title1LetterSpacing: 0.007,

    title2Family: '"Roboto", sans-serif',
    title2Size: 1.375,
    title2Weight: 400,
    title2LineHeight: 1.21428571,
    title2LetterSpacing: 0.008,

    title3Family: '"Roboto", sans-serif',
    title3Size: 1.25,
    title3Weight: 400,
    title3LineHeight: 1.27272727,
    title3LetterSpacing: 0.01,

    headlineFamily: '"Roboto", sans-serif',
    headlineSize: 1.0625,
    headlineWeight: 500,
    headlineLineHeight: 1.29411764,
    headlineLetterSpacing: -0.014,

    bodyFamily: '"Roboto", sans-serif',
    bodySize: 1.0625,
    bodyWeight: 400,
    bodyLineHeight: 1.29411764,
    bodyLetterSpacing: -0.012,

    calloutFamily: '"Roboto", sans-serif',
    calloutSize: 1,
    calloutWeight: 400,
    calloutLineHeight: 1.3125,
    calloutLetterSpacing: -0.01,

    subheadFamily: '"Roboto", sans-serif',
    subheadSize: 0.9375,
    subheadWeight: 400,
    subheadLineHeight: 1.33333333,
    subheadLetterSpacing: -0.008,

    footnoteFamily: '"Roboto", sans-serif',
    footnoteSize: 0.8125,
    footnoteWeight: 400,
    footnoteLineHeight: 1.38461538,
    footnoteLetterSpacing: -0.003,

    caption1Family: '"Roboto", sans-serif',
    caption1Size: 0.75,
    caption1Weight: 400,
    caption1LineHeight: 1.33333333,
    caption1LetterSpacing: 0,

    caption2Family: '"Roboto", sans-serif',
    caption2Size: 0.6875,
    caption2Weight: 400,
    caption2LineHeight: 1.181818181818,
    caption2LetterSpacing: 0.003,

    overlineFamily: '"Roboto", sans-serif',
    overlineSize: 0.6875,
    overlineWeight: 400,
    overlineLineHeight: 1.63644,
    overlineLetterSpacing: 0.075,
};
