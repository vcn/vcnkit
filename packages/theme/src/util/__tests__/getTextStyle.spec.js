import * as React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import styled from 'styled-components';

import fonts from '../../fonts';
import { getTextStyle } from '../../';

it('Provides hero style correctly', () => {
    const HeroText = styled.div`
        ${getTextStyle('hero')};
    `;

    const tree = renderer.create(<HeroText />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.heroSize}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.heroWeight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.heroLineHeight}`);
});

it('Provides headline style correctly', () => {
    const HeadlineText = styled.div`
        ${getTextStyle('headline')};
    `;

    const tree = renderer.create(<HeadlineText />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.headlineSize}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.headlineWeight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.headlineLineHeight}`);
});

it('Provides title1 style correctly', () => {
    const Title1Text = styled.div`
        ${getTextStyle('title1')};
    `;

    const tree = renderer.create(<Title1Text />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.title1Size}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.title1Weight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.title1LineHeight}`);
});

it('Provides title2 style correctly', () => {
    const Title2Text = styled.div`
        ${getTextStyle('title2')};
    `;

    const tree = renderer.create(<Title2Text />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.title2Size}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.title2Weight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.title2LineHeight}`);
});

it('Provides subheading1 style correctly', () => {
    const Subheading1Text = styled.div`
        ${getTextStyle('subheading1')};
    `;

    const tree = renderer.create(<Subheading1Text />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.subheading1Size}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.subheading1Weight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.subheading1LineHeight}`);
});

it('Provides subheading2 style correctly', () => {
    const Subheading2Text = styled.div`
        ${getTextStyle('subheading2')};
    `;

    const tree = renderer.create(<Subheading2Text />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.subheading2Size}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.subheading2Weight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.subheading2LineHeight}`);
});

it('Provides body1 style correctly', () => {
    const Body1Text = styled.div`
        ${getTextStyle('body1')};
    `;

    const tree = renderer.create(<Body1Text />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.body1Size}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.body1Weight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.body1LineHeight}`);
});

it('Provides body2 style correctly', () => {
    const Body2Text = styled.div`
        ${getTextStyle('body2')};
    `;

    const tree = renderer.create(<Body2Text />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.body2Size}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.body2Weight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.body2LineHeight}`);
});

it('Provides caption style correctly', () => {
    const CaptionText = styled.div`
        ${getTextStyle('caption')};
    `;

    const tree = renderer.create(<CaptionText />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('font-family', '"Roboto",sans-serif');
    expect(tree).toHaveStyleRule('font-size', `${fonts.captionSize}rem`);
    expect(tree).toHaveStyleRule('font-weight', `${fonts.captionWeight}`);
    expect(tree).toHaveStyleRule('line-height', `${fonts.captionLineHeight}`);
});
