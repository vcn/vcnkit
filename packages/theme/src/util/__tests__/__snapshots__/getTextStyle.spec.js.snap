// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provides body1 style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 400;
  font-size: 0.8125rem;
  line-height: 1.618;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides body2 style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.618;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides caption style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.618;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides headline style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.4;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides hero style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.4;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides subheading1 style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 400;
  font-size: 0.9375rem;
  line-height: 1.4;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides subheading2 style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 500;
  font-size: 0.9375rem;
  line-height: 1.4;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides title1 style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 400;
  font-size: 1.25rem;
  line-height: 1.4;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;

exports[`Provides title2 style correctly 1`] = `
.c0 {
  font-family: "Roboto",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.4;
  color: hsla(0,0%,0%,0.87);
}

<div
  className="c0"
/>
`;
